File: scharp/day/six/CSharpSpecFlowHealthcareExample/Domain/PatientDirectory.cs

namespace CSharpSpecFlowHealthcareExample.Domain
{
    public sealed class PatientDirectory
    {
        private readonly List<Patient> _patients = new();

        public void AddPatient(Patient patient)
        {
            ArgumentNullException.ThrowIfNull(patient);

            _patients.Add(patient);
        }

        public IReadOnlyList<Patient> SearchByLastName(
            string lastName)
        {
            ArgumentException.ThrowIfNullOrWhiteSpace(lastName);

            return _patients
                .Where(patient =>
                    patient.LastName.Equals(
                        lastName,
                        StringComparison.OrdinalIgnoreCase))
                .ToList();
        }
    }
}

File: java/java-practise/src/main/java/day/six/java-cucumber-healthcare/src/main/java/domain/PatientDirectory.java
package domain;

import java.util.ArrayList;
import java.util.List;
import java.util.Objects;

public final class PatientDirectory {
    private final List<Patient> patients =
            new ArrayList<>();

    public void addPatient(Patient patient) {
        patients.add(
                Objects.requireNonNull(
                        patient,
                        "patient must not be null"
                )
        );
    }

    public List<Patient> searchByLastName(
            String lastName
    ) {
        Objects.requireNonNull(
                lastName,
                "lastName must not be null"
        );

        return patients.stream()
                .filter(patient ->
                        patient.lastName()
                                .equalsIgnoreCase(lastName)
                )
                .toList();
    }
}
